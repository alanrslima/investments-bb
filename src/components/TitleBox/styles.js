import { StyleSheet } from 'react-native';
import { colors, fonts } from '../../styles';

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    padding: 20,
    justifyContent: 'space-between',
  },
  text: {
    color: colors.GRAY_DARK,
    fontWeight: '600',
    fontSize: fonts.DESCRIPTION,
  }
});
