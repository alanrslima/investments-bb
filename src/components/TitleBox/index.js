import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';

import styles from './styles';

function TitleBox({ title, subTitle }) {
  return (
    <View style={styles.container}>
      <Text style={styles.text}>{title}</Text>
      <Text style={styles.text}>{subTitle}</Text>
    </View>
  );
}

TitleBox.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string,
};

TitleBox.defaultProps = {
  subTitle: ''
};

export default TitleBox;
