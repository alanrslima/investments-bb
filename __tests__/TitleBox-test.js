import React from 'react';
import renderer from 'react-test-renderer';
import TitleBox from '../src/components/TitleBox';


test('renders correctly', () => {
  const tree = renderer.create(<TitleBox
    title="INVESTIMENTO"
    subTitle="R$"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
