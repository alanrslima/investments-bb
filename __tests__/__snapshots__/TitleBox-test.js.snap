// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "space-between",
      "padding": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#707070",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    INVESTIMENTO
  </Text>
  <Text
    style={
      Object {
        "color": "#707070",
        "fontSize": 15,
        "fontWeight": "600",
      }
    }
  >
    R$
  </Text>
</View>
`;
